internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -66,6 +67,36 @@ type HighPerformanceHooks struct {
 	sharedCPUs           string
 }
 
+func (h *HighPerformanceHooks) PreCreate(ctx context.Context, specgen *generate.Generator, s *sandbox.Sandbox, c *oci.Container) error {
+	log.Infof(ctx, "Run %q runtime handler pre-create hook for the container %q", HighPerformance, c.ID())
+	if !shouldRunHooks(ctx, c.ID(), specgen.Config, s) {
+		return nil
+	}
+
+	if requestedSharedCPUs(s.Annotations(), c.CRIContainer().GetMetadata().GetName()) {
+		if isContainerCPUsSpecEmpty(specgen.Config) {
+			return fmt.Errorf("no cpus found for container %q", c.Name())
+		}
+		cpusString := specgen.Config.Linux.Resources.CPU.Cpus
+		exclusiveCPUs, err := cpuset.Parse(cpusString)
+		if err != nil {
+			return fmt.Errorf("failed to parse container %q cpus: %w", c.Name(), err)
+		}
+		if h.sharedCPUs == "" {
+			return fmt.Errorf("shared CPUs were requested for container %q but none are defined", c.Name())
+		}
+		sharedCPUSet, err := cpuset.Parse(h.sharedCPUs)
+		if err != nil {
+			return fmt.Errorf("failed to parse shared cpus: %w", err)
+		}
+		// We must inject the environment variables in the PreCreate stage,
+		// because in the PreStart stage the process is already constructed.
+		// by the low-level runtime and the environment variables are presents.
```

Comment:
and the environment variables are already finalized

Reply:
Done.